apps/hash-graph/lib/graph/src/logging/init.rs

```diff
@@ -168,15 +169,15 @@ pub fn init_logger<P: AsRef<Path>>(
     };
 
     let json_file_appender =
-        tracing_appender::rolling::never(log_folder, format!("{log_file_name}.jsonl"));
+        tracing_appender::rolling::never(log_folder, format!("{log_file_prefix}.jsonl"));
     let (non_blocking, json_file_guard) = tracing_appender::non_blocking(json_file_appender);
 
     let json_file_layer = fmt::layer()
         .event_format(formatter.json())
         .fmt_fields(JsonFields::new())
         .with_writer(non_blocking);
 
-    let opentelemetry_layer = otlp_endpoint.map(configure_opentelemetry_layer);
+    let opentelemetry_layer = otlp_endpoint.as_deref().map(configure_opentelemetry_layer);
```

Comment:
Why do we need to `Deref` here and we previously didn't?

Reply:
Yes, previously we called `.as_deref()` before passing the argument to `init_logger`, since we take the args by struct now, it has moved here into the function. It's converting from `&Option<String>` to `Option<&str>`

Reply:
Ah, got it, thanks for clarifying!